src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -120,11 +124,15 @@ private ModifyQueueCapacityAction configureQueueCapacity(NodeKey esNode, Resourc
     return null;
   }
 
-  private int getNodeQueueCapacity(NodeKey esNode, ResourceEnum threadPool) {
-    // TODO: use NodeConfigurationRca to return capacity, for now returning defaults
-    if (threadPool.equals(ResourceEnum.SEARCH_THREADPOOL)) {
-      return 1000;
+  private int getNodeQueueCapacity(NodeKey esNode, Resource resource) {
```

Comment:
as I mentioned above, we don't have to change the input argument of this function. What we can do instead is to map queue rejection -> queue capacity